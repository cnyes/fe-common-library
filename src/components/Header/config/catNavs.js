const catNavs = [
  {
    name: 'myfunds',
    url: '/MyFunds.aspx',
    title: '我的基金',
    external: true,
    subItems: [
      {
        name: 'myInvestment',
        url: '/myInvestment/myInvestment.aspx?ga=nav',
        title: '投資組合分析',
        external: true,
      },
      {
        name: 'calcmyfundHTML5',
        url: '/calcmyfundHTML5.aspx?ga=nav',
        title: '基金透視鏡 ',
        external: true,
      },
    ],
  },
  {
    name: 'search',
    url: '/search/',
    title: '基金搜尋',
  },
  {
    name: 'ranking',
    url: '/ranking/index.htm?ga=nav',
    title: '基金排行',
    external: true,
    subItems: [
      {
        name: 'sector-ranking',
        url: '/sector-ranking/index.htm?ga=nav',
        title: '組別分類排行',
        external: true,
      },
      {
        name: 'sector-ranking-yearly',
        url: '/sector-ranking-yearly/index.htm?ga=nav',
        title: '組別年度績效排行',
        external: true,
      },
      {
        name: 'popylar-ranking',
        url: '/popular/index.htm?ga=nav',
        title: '熱門基金排行',
        external: true,
      },
    ],
  },
  {
    name: 'fixedincome',
    url: '/Fixedincome/',
    title: '債券專區',
    external: true,
    subItems: [
      {
        name: 'fixed-income',
        url: '/Fixedincome/index.aspx',
        title: '債券基金排行',
        external: true,
      },
      {
        name: 'fixed-income-search',
        url: '/Fixedincome/search.aspx',
        title: '債券基金搜尋器',
        external: true,
      },
    ],
  },
  {
    name: 'dividend',
    url: '/dividend/index.htm',
    title: '配息專區',
    external: true,
  },
  {
    name: 'report',
    url: '/report/report/index.htm',
    title: '研究報告',
    external: true,
    subItems: [
      {
        name: 'manager-view',
        url: '/report/manager/index.htm?ga=nav',
        title: '經理人訪談',
        external: true,
      },
      {
        name: 'fidelity-aasia-bond',
        url: 'http://campaign.cnyes.com/FidelityAsiaBond/?ga=nav',
        title: '亞債新視野',
        external: true,
      },
      {
        name: 'moring-star',
        url: '/report/MorningStar/index.htm?ga=nav',
        title: '晨星專欄',
        external: true,
      },
      {
        name: 'cnyes-radar',
        url: '/report/cnyes/index.htm?ga=nav',
        title: '鉅亨投資雷達',
        external: true,
      },
    ],
  },
  {
    name: 'fundsyes',
    url: 'https://www.fundsyes.com/?utm_source=cnyes&utm_medium=fundchannel_menu2',
    title: '基金交易平台',
    external: true,
  },
];

export default catNavs;
