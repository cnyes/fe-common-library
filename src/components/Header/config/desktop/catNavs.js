const catNavs = [
  {
    name: 'headline',
    url: '/news/cat/headline',
    title: '頭條',
  },
  {
    name: 'all',
    url: '/news/cat/all',
    title: '總覽',
  },
  {
    name: 'trending',
    url: '/trending',
    title: '人氣',
  },
  {
    name: 'tw_stock',
    url: '/news/cat/tw_stock',
    title: '台股',
  },
  {
    name: 'wd_stock',
    url: '/news/cat/wd_stock',
    title: '國際股',
  },
  {
    name: 'cn_stock',
    url: '/news/cat/cn_stock',
    title: 'A股港股',
  },
  {
    name: 'forex',
    url: '/news/cat/forex',
    title: '外匯',
  },
  {
    name: 'future',
    url: '/news/cat/future',
    title: '商品期貨',
  },
  {
    name: 'cnyeshouse',
    url: '/news/cat/cnyeshouse',
    title: '房產',
  },
  {
    name: 'tw_money',
    url: '/news/cat/tw_money',
    title: '理財',
  },
  {
    name: 'industry',
    url: '/news/cat/industry',
    title: '產業',
  },
  {
    name: 'event',
    url: '/news/cat/event',
    title: '時事',
  },
  {
    name: 'mag',
    url: '/news/cat/mag',
    title: '雜誌',
  },
  {
    name: 'celebrity_area',
    url: '/news/cat/celebrity_area',
    title: '鉅亨新視界',
  },
  {
    name: 'projects',
    url: '/projects',
    title: '專題報導',
  },
];

export default catNavs;
