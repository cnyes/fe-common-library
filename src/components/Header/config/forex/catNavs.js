const catNavs = [
  {
    name: 'news',
    url: 'https://news.cnyes.com/news/cat/forex',
    title: '新聞',
    external: true,
  },
  {
    name: 'forex',
    url: 'http://traderoom.cnyes.com/global/forex.aspx?ga=nav',
    title: '即時報價',
    external: true,
  },
  {
    name: 'forex_index',
    url: 'http://www.cnyes.com/forex/forex_index.aspx?ga=nav',
    title: '看盤室',
    external: true,
  },
  {
    name: 'major',
    url: 'http://www.cnyes.com/forex/major.aspx?ga=nav',
    title: '最新匯價',
    external: true,
  },
  {
    name: 'forex_list',
    url: 'http://www.cnyes.com/forex/forex_list.aspx?ga=nav',
    title: '交叉匯率',
    external: true,
  },
  {
    name: 'twd',
    url: 'http://www.cnyes.com/forex/twd.aspx?ga=nav',
    title: '新台幣',
    external: true,
  },
  {
    name: 'rmb',
    url: 'http://www.cnyes.com/forex/rmb.htm?ga=nav',
    title: '人民幣',
    external: true,
  },
  {
    name: 'jpy',
    url: 'http://www.cnyes.com/forex/jpy.htm?ga=nav',
    title: '日元',
    external: true,
  },
  {
    name: 'eur',
    url: 'http://www.cnyes.com/forex/eur.htm?ga=nav',
    title: '歐元',
    external: true,
  },
  {
    name: 'vote',
    url: 'http://www.cnyes.com/forex/vote.aspx?ga=nav',
    title: '多空調查',
    external: true,
  },
  {
    name: 'report',
    url: 'http://www.cnyes.com/forex/report.aspx?ga=nav',
    title: '研報',
    external: true,
  },
  {
    name: 'forexstat',
    url: 'http://www.cnyes.com/forex/forexStat.aspx?ga=nav',
    title: '績效統計',
    external: true,
  },
];

export default catNavs;
