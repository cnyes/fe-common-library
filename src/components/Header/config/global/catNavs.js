const catNavs = [
  {
    name: 'IndexImmediateQuotedPrice',
    url: 'http://www.cnyes.com/global/IndexImmediateQuotedPrice/?ga=nav',
    title: '國際股',
    external: true,
    subItems: [
      {
        name: 'wd_stock',
        url: 'https://news.cnyes.com/news/cat/wd_stock',
        title: '新聞',
        external: true,
      },
      {
        name: 'traderoom',
        url: 'http://traderoom.cnyes.com/global/index.aspx?ga=nav',
        title: '全球看盤',
        external: true,
      },
      {
        name: 'CountryAreaIndex',
        url: 'http://www.cnyes.com/global/CountryAreaIndex/Default.aspx?ga=nav',
        title: '國家&區域指數',
        external: true,
      },
      {
        name: 'IndexImmediateQuotedPrice_price',
        url: 'http://www.cnyes.com/global/IndexImmediateQuotedPrice/Default.aspx?ga=nav',
        title: '指數報價',
        external: true,
      },
      {
        name: 'IndexClosingQuotation_price1',
        url: 'http://www.cnyes.com/global/IndexClosingQuotation/Default.aspx?ga=nav',
        title: '指數收盤行情',
        external: true,
      },
      {
        name: 'NewHighLowIndex',
        url: 'http://www.cnyes.com/global/NewHighLowIndex/Default.aspx?ga=nav',
        title: '新高低指數',
        external: true,
      },
      {
        name: 'Rankings',
        url: 'http://www.cnyes.com/global/Rankings/Default.aspx?ga=nav',
        title: '指數排行',
        external: true,
      },
      {
        name: 'StockPrice',
        url: 'http://www.cnyes.com/global/StockPrice/Default.aspx?ga=nav',
        title: '類股報價',
        external: true,
      },
      {
        name: 'Technology',
        url: 'http://www.cnyes.com/global/Technology/Default.aspx?ga=nav',
        title: '技術選股',
        external: true,
      },
      {
        name: 'usainduststock',
        url: 'http://www.cnyes.com/usastock/usainduststock.aspx?ga=nav',
        title: '產業龍頭股',
        external: true,
      },
      {
        name: 'GlobalTime',
        url: 'http://www.cnyes.com/economy/indicator/GlobalTime/GlobalTime_Major.aspx?ga=nav',
        title: '股市交易時間',
        external: true,
      },
      {
        name: 'MarketInfo',
        url: 'http://www.cnyes.com/global/MarketInfo.aspx',
        title: '市場資訊',
        external: true,
      },
    ],
  },
  {
    name: 'usastock',
    url: 'http://www.cnyes.com/usastock/index.htm?ga=nav',
    title: '美股',
    external: true,
    subItems: [
      {
        name: 'us_stock',
        url: 'https://news.cnyes.com/news/cat/us_stock',
        title: '新聞',
        external: true,
      },
      {
        name: 'traderoom1',
        url: 'http://traderoom.cnyes.com/global/Stock.aspx',
        title: '看盤室',
        external: true,
      },
      {
        name: 'CountryAreaIndex',
        url: 'http://www.cnyes.com/global/CountryAreaIndex/Default.aspx?ga=nav',
        title: '自選股',
        external: true,
      },
      {
        name: 'ETF',
        url: 'http://www.cnyes.com/USASTOCK/ETF/ETFs.htm?ga=nav',
        title: 'ETFs',
        external: true,
      },
      {
        name: 'adr',
        url: 'http://www.cnyes.com/USASTOCK/adr/adrindex.aspx?ga=nav',
        title: 'ADR',
        external: true,
      },
      {
        name: 'sector',
        url: 'http://www.cnyes.com/USASTOCK/sector.aspx?ga=nav',
        title: '類股',
        external: true,
      },
      {
        name: 'usastock_stock',
        url: 'http://www.cnyes.com/USASTOCK/usastock_stock.aspx?ga=nav',
        title: '個股',
        external: true,
      },
      {
        name: 'Screener',
        url: 'http://www.cnyes.com/USASTOCK/Rankings/Screener.aspx?ga=nav',
        title: '排行榜',
        external: true,
      },
      {
        name: 'Vote',
        url: 'http://www.cnyes.com/USASTOCK/Vote.aspx?ga=nav',
        title: '多空調查',
        external: true,
      },
      {
        name: 'report',
        url: 'http://www.cnyes.com/USASTOCK/report/report.aspx?ga=nav',
        title: '研究報告',
        external: true,
      },
      {
        name: 'steatement',
        url: 'http://www.cnyes.com/USASTOCK/UsaSchedule/steatement.aspx?ga=nav',
        title: '行事曆',
        external: true,
      },
    ],
  },
  {
    name: 'hkstock_index',
    url: 'http://www.cnyes.com/hkstock/index.htm?ga=nav',
    title: '港股',
    external: true,
    subItems: [
      {
        name: 'traderoom1',
        url: 'http://traderoom.cnyes.com/global/index.aspx?ga=nav',
        title: '看盤室',
        external: true,
      },
      {
        name: 'my_hkstock',
        url: 'http://www.cnyes.com/personal/mycnyes/my_hkstock.asp?pagetype=hkstock&ga=nav',
        title: '自選股',
        external: true,
      },
      {
        name: 'hk_stock',
        url: 'https://news.cnyes.com/news/cat/hk_stock',
        title: '新聞',
        external: true,
      },
      {
        name: 'hkcode',
        url: 'http://www.cnyes.com/hkstock/hkcode.aspx?ga=nav',
        title: '類股',
        external: true,
      },
      {
        name: 'gb',
        url: 'http://www.cnyes.com/hkstock/gb.aspx?ga=nav',
        title: '個股',
        external: true,
      },
      {
        name: 'warrant_list',
        url: 'http://www.cnyes.com/hkstock/warrant_list.aspx?ga=nav',
        title: '權證',
        external: true,
      },
      {
        name: 'warrantSec',
        url: 'http://www.cnyes.com/hkstock/warrantSec.aspx?ga=nav',
        title: '權證搜尋',
        external: true,
      },
      {
        name: 'rank_up',
        url: 'http://www.cnyes.com/hkstock/rank_up.aspx?ga=nav',
        title: '排行榜',
        external: true,
      },
      {
        name: 'ipo_news',
        url: 'http://www.cnyes.com/hkstock/ipo_news.aspx?ga=nav',
        title: '新股追蹤',
        external: true,
      },
      {
        name: 'hkvote',
        url: 'http://www.cnyes.com/hkstock/hkvote.aspx?ga=nav',
        title: '多空調查',
        external: true,
      },
      {
        name: 'rsh_stock_wd',
        url: 'http://www.cnyes.com/report/rsh_stock_wd.aspx?ga=nav',
        title: '研究報告',
        external: true,
      },
      {
        name: 'tvonline',
        url: 'http://www.cnyes.com/hkstock/tvonline.htm?ga=nav',
        title: '直播節目',
        external: true,
      },
      {
        name: 'class',
        url: 'http://www.cnyes.com/hkstock/class/?ga=nav',
        title: '港股教室',
        external: true,
      },
      {
        name: 'hkCompany_Cal',
        url: 'http://www.cnyes.com/hkstock/hkCompany_Cal.aspx?mtype=TODAY&ga=nav',
        title: '行事曆',
        external: true,
      },
    ],
  },
  {
    name: 'shstock_index',
    url: 'http://www.cnyes.com/shstock/index.htm?ga=nav',
    title: '滬深股',
    external: true,
    subItems: [
      {
        name: 'shstock',
        url: 'http://www.cnyes.com/shstock/index.htm?ga=nav',
        title: '滬深股首頁',
        external: true,
      },
      {
        name: 'Astock',
        url: 'http://campaign.cnyes.com/Astock/',
        title: '投資講座',
        external: true,
      },
      {
        name: 'Analytics_Tech',
        url: 'http://www.cnyes.com/shstock/Analytics_Tech.aspx?ga=nav',
        title: '選股分析',
        external: true,
      },
      {
        name: 'search',
        url: 'http://www.cnyes.com/shstock/search.htm?ga=nav',
        title: '個股搜尋',
        external: true,
      },
      {
        name: 'ranking1',
        url: 'http://www.cnyes.com/shstock/ranking.aspx?ga=nav',
        title: '績效排行',
        external: true,
      },
      {
        name: 'mainindex',
        url: 'http://www.cnyes.com/shstock/mainindex.aspx?ga=nav',
        title: '指數查詢',
        external: true,
      },
      {
        name: 'newstock',
        url: 'http://www.cnyes.com/shstock/newstock.aspx?ga=nav',
        title: '新股',
        external: true,
      },
      {
        name: 'industry',
        url: 'http://www.cnyes.com/shstock/industry.htm?ga=nav',
        title: '類股排行',
        external: true,
      },
      {
        name: 'mystock1',
        url: 'http://traderoom.cnyes.com/global/mystock.aspx?ga=nav',
        title: '看盤室',
        external: true,
      },
      {
        name: 'lkc',
        url: 'http://www.cnyes.com/shstock/lkc.aspx?ga=nav',
        title: '李克強指標',
        external: true,
      },
    ],
  },
  {
    name: 'jp_index',
    url: 'http://www.cnyes.com/JP/index.htm?ga=nav',
    title: '日股',
    external: true,
    subItems: [
      {
        name: 'sande_1',
        url: 'http://www.cnyes.com/JP/sande_1.aspx?ga=nav',
        title: '股匯市表現',
        external: true,
      },
      {
        name: 'maindex1',
        url: 'http://www.cnyes.com/JP/maindex1.aspx?ga=nav',
        title: '日股指數',
        external: true,
      },
      {
        name: 'nikkei225',
        url: 'http://www.cnyes.com/JP/nikkei225.aspx?ga=nav',
        title: '日經225成分股',
        external: true,
      },
      {
        name: 'famous',
        url: 'http://www.cnyes.com/JP/famous.aspx?ga=nav',
        title: '知名企業',
        external: true,
      },
      {
        name: 'jp_stock',
        url: 'http://traderoom.cnyes.com/global/stock.aspx?ga=nav',
        title: '日股看盤室',
        external: true,
      },
      {
        name: 'fund',
        url: 'https://fund.cnyes.com/',
        title: '日本基金',
        external: true,
      },
      {
        name: 'jpy',
        url: 'http://www.cnyes.com/forex/jpy.htm?ga=nav',
        title: '日元',
        external: true,
      },
    ],
  },
  {
    name: 'futures_index',
    url: 'http://www.cnyes.com/futures/index.htm?ga=nav',
    title: '國際期貨',
    external: true,
    subItems: [
      {
        name: 'future_news',
        url: 'https://news.cnyes.com/news/cat/future',
        title: '商品新聞',
        external: true,
      },
      {
        name: 'traderoom_futures',
        url: 'http://traderoom.cnyes.com/global/futures.aspx?ga=nav',
        title: '看盤室',
        external: true,
      },
      {
        name: 'indexftr',
        url: 'http://www.cnyes.com/futures/indexftr.aspx?ga=nav',
        title: '指數期貨',
        external: true,
      },
      {
        name: 'price',
        url: 'http://www.cnyes.com/futures/price.aspx?ga=nav',
        title: '商品期貨',
        external: true,
      },
      {
        name: 'rank1',
        url: 'http://www.cnyes.com/futures/rank1.aspx?ga=nav',
        title: '商品績效排行',
        external: true,
      },
      {
        name: 'energy',
        url: 'http://www.cnyes.com/futures/energy.aspx?ga=nav',
        title: '能源',
        external: true,
      },
      {
        name: 'heavymetal',
        url: 'http://www.cnyes.com/futures/heavymetal.aspx?ga=nav',
        title: '貴重金屬',
        external: true,
      },
      {
        name: 'basicmetal',
        url: 'http://www.cnyes.com/futures/basicmetal.aspx?ga=nav',
        title: '基本金屬',
        external: true,
      },
      {
        name: 'agricultural',
        url: 'http://www.cnyes.com/futures/agricultural.aspx?ga=nav',
        title: '農產品',
        external: true,
      },
      {
        name: 'material',
        url: 'http://www.cnyes.com/futures/material.aspx?ga=nav',
        title: '原物料',
        external: true,
      },
      {
        name: 'contract',
        url: 'http://www.cnyes.com/futures/contract.aspx?ga=nav',
        title: '商品合約規格',
        external: true,
      },
    ],
  },
  {
    name: 'presh_index',
    url: 'http://www.cnyes.com/presh/index.htm?ga=nav',
    title: '債券',
    external: true,
    subItems: [
      {
        name: 'futu_bond',
        url: 'https://news.cnyes.com/news/cat/futu_bond',
        title: '債券新聞',
        external: true,
      },
      {
        name: 'twBondMarket1',
        url: 'http://www.cnyes.com/bond/twBondMarket1.aspx?ga=nav',
        title: '台灣債券',
        external: true,
      },
      {
        name: 'intBondMarket1',
        url: 'http://www.cnyes.com/bond/intBondMarket1.aspx?ga=nav',
        title: '國際債券',
        external: true,
      },
      {
        name: 'intLoan1',
        url: 'https://www.cnyes.com/?aspxerrorpath=/bond/intLoan1.aspx',
        title: '國際拆放款',
        external: true,
      },
      {
        name: 'eurodebt',
        url: 'http://www.cnyes.com/eurodebt.htm?ga=nav',
        title: '歐洲PIGS債券',
        external: true,
      },
      {
        name: 'bondCalculationJS',
        url: 'http://www.cnyes.com/bond/bondCalculationJS.aspx?ga=nav',
        title: '債券試算',
        external: true,
      },
      {
        name: 'bondClass1',
        url: 'http://www.cnyes.com/bond/bondClass1.aspx?ga=nav',
        title: '債券教室',
        external: true,
      },
    ],
  },
  {
    name: 'gold',
    url: 'http://www.cnyes.com/gold/index.htm?ga=nav',
    title: '黃金',
    external: true,
    subItems: [
      {
        name: 'precious_metals',
        url: 'https://news.cnyes.com/news/cat/precious_metals',
        title: '黃金新聞',
        external: true,
      },
      {
        name: 'precious_metals1',
        url: 'https://news.cnyes.com/news/cat/precious_metals',
        title: '黃金分析',
        external: true,
      },
      {
        name: 'GoldFund',
        url: 'http://www.cnyes.com/gold/GoldFund.aspx?ga=nav',
        title: 'ETFs/基金',
        external: true,
      },
      {
        name: 'GoldStock',
        url: 'http://www.cnyes.com/gold/GoldStock.aspx?ga=nav',
        title: '相關個股',
        external: true,
      },
      {
        name: 'Vote',
        url: 'http://www.cnyes.com/gold/Vote.aspx?ga=nav',
        title: '多空調查',
        external: true,
      },
      {
        name: 'ClassRoom',
        url: 'http://www.cnyes.com/gold/ClassRoom.aspx?ga=nav',
        title: '黃金教室',
        external: true,
      },
      {
        name: 'SearchBlog',
        url: 'http://blog.cnyes.com/SearchBlog.aspx?Purpose=blog&ClassifyId=15&ga=nav',
        title: '黃金Blog',
        external: true,
      },
      {
        name: 'Taifex',
        url: 'http://www.cnyes.com/gold/Taifex.aspx?ga=nav',
        title: '黃金合約',
        external: true,
      },
      {
        name: 'finance',
        url: 'http://www.cnyes.com/chn/finance/index.htm?ga=nav',
        title: '金融中心',
        external: true,
      },
    ],
  },
  {
    name: 'CentralBank',
    url: 'http://www.cnyes.com/CentralBank/index.htm?ga=nav',
    title: '全球央行',
    external: true,
    subItems: [
      {
        name: 'interest1',
        url: 'http://www.cnyes.com/CentralBank/interest1.htm?ga=nav',
        title: '基準利率',
        external: true,
      },
      {
        name: 'intLoan',
        url: 'http://www.cnyes.com/CentralBank/intLoan.htm?ga=nav',
        title: '拆款利率',
        external: true,
      },
      {
        name: 'exchange',
        url: 'http://www.cnyes.com/CentralBank/exchange.htm?ga=nav',
        title: '外匯存底',
        external: true,
      },
      {
        name: 'SovRatingAll',
        url: 'http://www.cnyes.com/CentralBank/SovRatingAll.htm?ga=nav',
        title: '主權評級',
        external: true,
      },
      {
        name: 'fStatistics1',
        url: 'http://www.cnyes.com/CentralBank/fStatistics1.htm?ga=nav',
        title: '金融統計',
        external: true,
      },
      {
        name: 'goldResere',
        url: 'http://www.cnyes.com/CentralBank/goldResere.htm?ga=nav',
        title: '黃金儲備',
        external: true,
      },
      {
        name: 'moneyPolicy',
        url: 'http://www.cnyes.com/CentralBank/moneyPolicy.htm?ga=nav',
        title: '貨幣政策',
        external: true,
      },
      {
        name: 'curCodity',
        url: 'http://www.cnyes.com/CentralBank/curCodity.htm?ga=nav',
        title: '貨幣商品',
        external: true,
      },
      {
        name: 'cenBanks',
        url: 'http://www.cnyes.com/CentralBank/cenBanks.htm?ga=nav',
        title: '各國央行',
        external: true,
      },
      {
        name: 'GlobalData_Major',
        url: 'http://www.cnyes.com/economy/indicator/Page/GlobalData_Major.aspx?ga=nav',
        title: '指標透試鏡',
        external: true,
      },
      {
        name: 'govSpeech',
        url: 'http://www.cnyes.com/fc/govSpeech.asp?govtype=5&pagetype=finance&ga=nav',
        title: '舊版央行',
        external: true,
      },
    ],
  },
  {
    name: 'schedule',
    url: 'http://www.cnyes.com/economy/indicator/Page/schedule.aspx?ga=nav',
    title: '經濟指標',
    external: true,
    subItems: [
      {
        name: 'macro',
        url: 'http://www.cnyes.com/economy/macro/index.htm?ga=nav',
        title: '宏觀',
        external: true,
      },
      {
        name: 'EconomicsCalendar',
        url: 'http://www.cnyes.com/economy/indicator/EconomicsCalendar/Calendar.aspx?ga=nav',
        title: '金融行事曆',
        external: true,
      },
      {
        name: 'risk',
        url: 'http://www.cnyes.com/risk.htm?ga=nav',
        title: '風險指標',
        external: true,
      },
      {
        name: 'luxury',
        url: 'http://www.cnyes.com/luxury.htm?ga=nav',
        title: '奢侈品',
        external: true,
      },
    ],
  },
  {
    name: 'StockQ',
    url: 'http://money.cnyes.com/StockQ.aspx?ga=nav',
    title: 'StockQ',
    external: true,
  },
];

export default catNavs;
