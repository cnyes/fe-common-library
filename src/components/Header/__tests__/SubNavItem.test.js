/* eslint-disable import/first,func-names,global-require,prefer-arrow-callback */
import React from 'react';
import { mount } from 'enzyme';
import catNavs from '../config/fund/catNavs';

describe('<SubNavItem />', function() {
  let SubNavItem;
  let makeSubject;
  const mockParams = catNavs[0];

  beforeEach(() => {
    jest.resetModules();

    global.__SERVER__ = false;
    global.__CLIENT__ = true;

    SubNavItem = require('../SubNavItem');

    makeSubject = params => {
      const props = params || mockParams;

      return mount(<SubNavItem {...props} />);
    };
  });

  it('should make a snapshot', () => {
    const subject = makeSubject();

    expect(subject.html()).toMatchSnapshot();
  });

  it('should be nice if !external', () => {
    const subject = makeSubject(catNavs[1]);

    expect(subject.props().external).not.toEqual(true);
    expect(subject.find('Link').length).toEqual(1);
  });

  it('should be nice if external', () => {
    const subject = makeSubject({
      ...catNavs[0],
      subItems: null,
    });

    expect(subject.props().external).toEqual(true);
    expect(subject.find('Link').length).toEqual(0);
  });

  describe('SubNavItemMenu', () => {
    it('should not render if undefined', () => {
      const subject = makeSubject(catNavs[1]);

      expect(subject.props().subItems).not.toBeDefined();
      expect(subject.find('nav').length).toEqual(0);
    });

    it('should not render if []', () => {
      const subject = makeSubject({
        ...catNavs[1],
        subItems: [],
      });

      expect(subject.props().subItems).toEqual([]);
      expect(subject.find('nav').length).toEqual(0);
    });

    it('should be nice', () => {
      const subject = makeSubject(catNavs[0]);

      expect(subject.props().subItems).not.toEqual([]);
      expect(subject.find('nav').length).toEqual(1);
      expect(subject.find('nav a').length).toEqual(catNavs[0].subItems.length);
    });
  });
});
