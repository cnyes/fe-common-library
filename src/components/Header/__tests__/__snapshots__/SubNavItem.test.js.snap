// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubNavItem /> should make a snapshot 1`] = `"<span class=\\"cat-nav-item\\"><a class=\\"with-arrow\\" href=\\"/MyFunds.aspx\\" title=\\"我的基金\\">我的基金</a><nav class=\\"cat-nav-sub-item\\"><a href=\\"/myInvestment/myInvestment.aspx?ga=nav\\" title=\\"投資組合分析\\">投資組合分析</a><a href=\\"/calcmyfundHTML5.aspx?ga=nav\\" title=\\"基金透視鏡 \\">基金透視鏡 </a></nav></span>"`;
